vnext/.npmignore

```diff
@@ -1,42 +1 @@
-.eslintignore
-.eslintrc.js
-.vs
-.vscode
-**/*.inc
-**/*.windesktop.*
-api-extractor.json
-build
-CMakeLists.txt
-/Desktop
-Desktop.Dll
-Desktop.ABITests
-Desktop.IntegrationTests
-Desktop.Test.DLL
-Desktop.UnitTests
-FollyWin32
-IntegrationTests
-IntegrationTestScripts
-JSI.Desktop.UnitTests
-/NuGet.Config
-Microsoft.ReactNative.Cxx.UnitTests
-Microsoft.ReactNative.IntegrationTests
-Microsoft.ReactNative.Managed.UnitTests
-package-deps.json
-packages
-react-native-windows.build.log
-ReactCopies/
-ReactWindows-Universal.sln
-ReactWindows-Desktop.sln
-RNTester.*
-RNTester/
-Mso.UnitTests
-proposals
-src/
-target
-temp
-tsconfig.json
-Universal.IntegrationTests
-Universal.UnitTests
-win32
-WindowsSampleCSharpApp
-yarn.lock
+# This project follows an explicit 'include' pattern as defined in the `files` section of pacakge.json
```

Comment:
nit: consider just removing this file

Reply:
I thought to leave the file and keep it empty with a comment so others won't fall into the trap of re-adding it, and now it is clearly documented for reviewers in the future as well.

I'll also have to retest everything because with this file gone npm pack could start picking up patterns from `.gitignore`.

Reply:
Makes sense